### PR TITLE
[Snackbar] Exposed the snackbar button class

### PR DESCRIPTION
**Pull request by yarneo**
Exposed the snackbar button class in the manager to allow more fine tuning theming that isn't color/typography related.

### CODE REVIEW

File: components/Snackbar/src/MDCSnackbarManager.h [hunk 1]
```diff
@@ -147,6 +147,13 @@
  */
 @property(class, nonatomic, strong, nullable) UIFont *buttonFont;
 
+/**
+ The button class to set the snackbar button.
+ Note: The class must be a subclass of MDCSnackbarMessageViewButton.
+
+ Default is set to MDCSnackbarMessageViewButton.
+ */
+@property(class, nonatomic, strong, nullable) Class buttonClass;
```
Comment (ianegordon) [inline components/Snackbar/src/MDCSnackbarManager.h]: nonnull?

**Review by ianegordon — DISMISSED** [review]

**Review by randallli — APPROVED** [review]

File: components/Snackbar/src/MDCSnackbarManager.h [hunk 2]
```diff
@@ -147,6 +147,13 @@
  */
 @property(class, nonatomic, strong, nullable) UIFont *buttonFont;
 
+/**
+ The button class to set the snackbar button.
+ Note: The class must be a subclass of MDCSnackbarMessageViewButton.
+
+ Default is set to MDCSnackbarMessageViewButton.
+ */
```
Comment (jverkoey) [inline components/Snackbar/src/MDCSnackbarManager.h]: I wonder whether a delegate pattern would be cleaner here. I’m hesitant to add an api that encourages subclassing for what is likely just stylistic needs.

**Review by jverkoey — DISMISSED** [review]
Quick feedback while waiting for food.

File: components/Snackbar/src/MDCSnackbarManager.h (hunk 2 above)
Comment (yarneo) [reply components/Snackbar/src/MDCSnackbarManager.h]: moved to a delegate pattern.

File: components/Snackbar/src/MDCSnackbarMessageView.h [hunk 3]
```diff
@@ -101,3 +107,6 @@ __deprecated_msg("Use messsageTextColor instead.");
 
 @end
 // clang-format on
+
+@interface MDCSnackbarMessageViewButton : MDCFlatButton
```
Comment (jverkoey) [inline components/Snackbar/src/MDCSnackbarMessageView.h]: Can we keep this as a private implementation detail? I see you're exposing MDCButton instances above so am wondering if we can continue hiding the subclass implementation.

**Review by jverkoey — CHANGES_REQUESTED** [review]
Looks good overall - one small question about exposing the private button implementation.

File: components/Snackbar/src/MDCSnackbarMessageView.h (hunk 3 above)
Comment (yarneo) [reply components/Snackbar/src/MDCSnackbarMessageView.h]: done. the idea behind this is if snackbar has specific properties for its button in the future, it won't be exposed and we won't be able to theme those properties.

**Review by jverkoey — APPROVED** [review]

**Pull request merged**